app/presentation/panels/main_panel.py

```diff
@@ -118,27 +128,47 @@ def handle_connection_prior(self, connect: Callable[[str], ServiceError]):
                                    informative_text=message,
                                    notification_variant=NotificationVariant.Error)
 
-    def setup_button_actions(self,
-                             calibration: Callable[[int, int], ServiceError],
-                             laser_write: Callable[[List[List[Tuple[int, int]]], str, bool], None],
-                             prior_init: Callable[[str], ServiceError],
-                             laser_init: Callable[[str], ServiceError],
-                             stage_info: Callable[[], List]):
+    def setup_actions(self, calibration: Callable[[int, int], ServiceError],
+                      laser_write: Callable[[List[List[Tuple[int, int]]], str, bool, int], None],
+                      prior_init: Callable[[str], ServiceError],
+                      laser_init: Callable[[str], ServiceError],
+                      stage_info: Callable[[], List],
+                      draw_file_preview: Callable[[bool, str, Canvas, int], None]):
 
         self.stage_management_grid.button_calibration.clicked.connect(
             lambda: Thread(target=self.handle_calibration_result, args=(calibration,), daemon=True).start()
         )
         self.stage_management_grid.button_start.clicked.connect(
-            lambda: laser_write(self.canvas.get_points, self.stage_management_grid.get_selected_file,
-                                self.stage_management_grid.check_box.isChecked())
+            lambda: laser_write(self.canvas.get_points,
+                                self.canvas.get_scaled_figures,
+                                self.stage_management_grid.from_canvas_checkbox.isChecked(),
+                                self.print_scale_slider.value())
         )
+        self.port_coms_grid.button_connect_stage.clicked.connect(
+            lambda: self.handle_connection_prior(prior_init)
+        )
+
+        def on_slider_value_changed(value: int):
+            self.update_print_scale_label(value)
+            draw_file_preview(
+                self.stage_management_grid.from_canvas_checkbox.isChecked(),
+                self.stage_management_grid.get_selected_file,
+                self.canvas,
+                value
+            )
+
+        self.slider.print_scale_slider.valueChanged.connect(on_slider_value_changed)
+
+        self.stage_info_grid.start_timer(stage_info)
+
         self.port_coms_grid.button_connect_stage.clicked.connect(
             lambda: self.handle_connection_prior(prior_init)
         )
         self.port_coms_grid.button_connect_laser.clicked.connect(
             lambda: self.handle_connection_laser(laser_init)
         )
-        self.stage_info_grid.start_timer(stage_info) # TODO think later about how to run it in different thread
+
+        self.stage_management_grid.button_load_file.clicked.connect(lambda x: WindowUtils.open_file(self))
 
     @staticmethod
     def show_notification(title: str = "SUCCESS",
```

Comment:
There is a huge problem with notifications, as whem they appear they clear all canvas.